village/src/App.js

```diff
@@ -11,14 +13,37 @@ class App extends Component {
       smurfs: [],
     };
   }
+  componentDidMount() {
+    axios.get("http://localhost:3333/smurfs")
+    .then(res => {
+      this.setState({ smurfs: res.data });
+    })
+    .catch(err => console.log(err));
+  }
+  newSmurfList = data => {
+    this.setState({
+      smurfs: data
+    });
+  };
   // add any needed code to ensure that the smurfs collection exists on state and it has data coming from the server
   // Notice what your map function is looping over and returning inside of Smurfs.
   // You'll need to make sure you have the right properties on state and pass them down to props.
   render() {
     return (
       <div className="App">
-        <SmurfForm />
+        <nav>
+          <NavLink to="/">Smurf Village</NavLink>
+          <NavLink to="/smurf-form">Add a Smurf</NavLink>
+        </nav>
+        <Route
+            exact
+            path="/smurf-form"
+            render={props => (
+              <SmurfForm {...props} newSmurfList={this.newSmurfList} />
+            )}
+          />
         <Smurfs smurfs={this.state.smurfs} />
+        <Footer />
```

Comment:
Technically, you will want to make the smurfs component here the home route so that if you're application gets bigger, you can choose if it gets rendered or not. As a component in App.js that is not defined in a route, it is always going to show. Because you are passing props to it, it will need to use render prop in the route like so, <Route exact path="/" render={props => <Smurfs {...props} smurfs={this.state.smurfs} />} />